### PR TITLE
Add check for length of call data in transaction

### PR DESCRIPTION
**Pull request by tcoratger**
Time spent on this PR: 30 min

Resolves: #976

# Pull Request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

- During the transformation of the transaction to a Starknet transaction, before filling the calldata, a check is made to know its size and return an error if this size exceeds the environment variable provided (`MAX_FELTS_IN_CALLDATA`).
- A unit test is added to check the proper implementation of this feature.

# Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/eth_provider/error.rs [hunk 1]
```diff
@@ -51,6 +51,9 @@ pub enum EthApiError {
     /// Other Kakarot error
     #[error("kakarot error: {0}")]
     KakarotError(KakarotError),
+    /// Error related to transaction calldata being too large.
+    #[error("too large calldata")]
+    TooLargCallData,
```
Comment (Eikix) [inline src/eth_provider/error.rs:56]: maybe TooLargeCalldata?

**Review by Eikix — COMMENTED** [review]
works for me!

File: src/eth_provider/error.rs (hunk 1 above)
Comment (tcoratger) [reply src/eth_provider/error.rs]: Oups yes just fixed

**Review by Eikix — APPROVED** [review]
lgtm

File: src/eth_provider/error.rs [hunk 2]
```diff
@@ -51,6 +51,9 @@ pub enum EthApiError {
     /// Other Kakarot error
     #[error("kakarot error: {0}")]
     KakarotError(KakarotError),
+    /// Error related to transaction calldata being too large.
+    #[error("too large calldata")]
+    TooLargeCallData,
```
Comment (greged93) [inline src/eth_provider/error.rs:56]:
```suggestion
    #[error("calldata exceeded limit of {0}: {1}")]
    CalldataExceededLimit(u64, u64),
```

File: src/eth_provider/starknet/kakarot_core.rs [hunk 3]
```diff
@@ -135,3 +148,36 @@ pub fn to_starknet_transaction(
...
+                // Print a message indicating that the call data is too large
+                println!("Calldata too large error confirmed");
+            }
+            // If the result is any other error or a successful result
+            _ => {
+                // Panic with a message indicating that an error indicating too large call data was expected
+                panic!("Expected error EthApiError::TooLargeCallData");
+            }
+        }
+    }
```
Comment (greged93) [inline src/eth_provider/starknet/kakarot_core.rs:182]: you can unwrap here and add `#[should_panic(expected = "too large calldata")]` under `#[test]` (or the new error message if you update it).

**Review by greged93 — CHANGES_REQUESTED** [review]
requested changes before merge

File: src/eth_provider/starknet/kakarot_core.rs (hunk 3 above)
Comment (tcoratger) [reply src/eth_provider/starknet/kakarot_core.rs]: Yes, just fixed

**Review by greged93 — APPROVED** [review]

**Pull request merged**